src/Mod/Assembly/App/AssemblyObject.cpp

```diff
@@ -1833,7 +1870,11 @@ bool AssemblyObject::isEdgeType(App::DocumentObject* obj,
                                 std::string& elName,
                                 GeomAbs_CurveType type)
 {
-    PartApp::Feature* base = static_cast<PartApp::Feature*>(obj);
+    auto* base = dynamic_cast<PartApp::Feature*>(obj);
```

Comment:
What about comparing typeid instead of dynamic cast? Better to use if this function will be called a lot

Reply:
Not a bad idea -- but I'm going to consider it "future work" for now.